sdk/src/Core/Amazon.Runtime/Telemetry/TelemetryProvider.cs

```diff
@@ -27,25 +28,41 @@ public abstract class TelemetryProvider
         /// <summary>
         /// Gets the <see cref="MeterProvider"/> used to create new metrics.
         /// </summary>
-        public abstract MeterProvider MeterProvider { get; protected set; }
+        public MeterProvider MeterProvider { get; private set; }
 
         /// <summary>
         /// Gets the <see cref="TracerProvider"/> used to create new tracers.
         /// </summary>
-        public abstract TracerProvider TracerProvider { get; protected set; }
+        public TracerProvider TracerProvider { get; private set; }
+
+        public TelemetryProvider(MeterProvider meterProvider, TracerProvider tracerProvider)
+        {
+            MeterProvider = meterProvider;
+            TracerProvider = tracerProvider;
```

Comment:
Should this be calling the Register* methods instead so it does the null check or is it valid to be able to specify null here but not if the Register* methods are called?